cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -135,6 +136,16 @@
     private KafkaAuthorization authorization;
     private SortedMap<Integer, String> externalAddresses = new TreeMap<>();
 
+    // Templates
+    protected Map<String, String> templateExternalBootstrapServiceLabels;
+    protected Map<String, String> templateExternalBootstrapServiceAnnotations;
+    protected Map<String, String> templatePerPodServiceLabels;
+    protected Map<String, String> templatePerPodServiceAnnotations;
+    protected Map<String, String> templateExternalBootstrapRouteLabels;
+    protected Map<String, String> templateExternalBootstrapRouteAnnotations;
+    protected Map<String, String> templatePerPodRouteLabels;
+    protected Map<String, String> templatePerPodRouteAnnotations;
```

Comment:
Something that Standa started to use in the `KafkaMirrorMaker` (with consumer and producer spec) not saving all these variables but just a `KafkaClusterTemplate template` field. Wdyt?

Reply:
Thinking more, maybe too many places and times where checking if template field is null or not (so the solution could not be good).

Reply:
Yeah, that is exactly what I did at first, stored the whole template. But as you say, this would mean that through the code you need to check for null all the time. I think it is a bit better to do it all in one place in `fromCrd`.

What I was thinking about was if we should not set the defaults already in the API (in the model classes) - in this case the default could be an actual Template object with empty maps for labels and annotations. But this doesn't work either because for example the whole `entityOperator` part might be missing in the Kafka cluster CR. So I think the current way is the best.